lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -368,13 +365,25 @@ async sub transform_transaction {
             timestamp    => $int_timestamp,
         );
 
-        my @transactions = await $self->_check_contract_transaction($transaction, $receipt);
-        push @transactions, $transaction if ($amount->bgt(0));
-
         if (!$self->accounts() || ($self->latest_accounts_update + UPDATE_ACCOUNTS <= time)) {
             await $self->get_hash_accounts();
         }
 
+        # if the gas is empty we don't proceed
+        unless ($gas && $decoded_transaction->{gasPrice}) {
+            my $our_transaction = await $self->_set_transaction_type($transaction);
```

Comment:
this is not clear to me, that is no relation between the set_transaction_type method and the missing receipt and gasprice, this seems unnecessary at this point, better to remove, also I don't see why are we moving these checks to this place, the gasprice check can be done in the same place as before

Reply:
@reginaldo-binary i used `set_transaction_type` here to make sure that logs are printed for our own transactions only. I believe we receive all transactions from the block-chain so we should not log for every transaction when `gasUsed` is empty.